simexpal/build.py

```diff
@@ -271,7 +272,12 @@ def git_ref_changed():
 				# Clone the git repository into the build.source_dir
 				subprocess.check_call(['git', 'clone', build.info.git_repo, build.source_dir])
 
-			util.touch(os.path.join(checkout_dir, 'checkedout.simexpal'))
+			# Mark checkout as finished
+			with open(cfg.basedir + '/simex.cache', 'r') as myfile:
+				cache = json.load(myfile)
+				cache["checkout_done"] = True
+			with open(cfg.basedir + '/simex.cache', 'w') as myfile:
```

Comment:
Again. The rw access in two different scopes. I am wondering if there is perhaps a benefit of reading and writing in two different scopes? Is the limit in access beneficial to the separation of concern?